src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,30 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserException("Incorrect value");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("A user with this login is already registered");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new InvalidUserException("Login must contain at least 6 characters");
```

Comment:
This check for login length should come after the null check for login to avoid potential NullPointerException.